astropy/stats/tests/test_funcs.py

```diff
@@ -440,6 +442,42 @@ def test_mad_std():
         data = normal(5, 2, size=(100, 100))
         assert_allclose(funcs.mad_std(data), 2.0, rtol=0.05)
 
+def test_mad_std_scalar_return():
+    with NumpyRNGContext(12345):
+        data = normal(5, 2, size=(10, 10))
+        # make a masked array with no masked points
+        data = np.ma.masked_where(np.isnan(data), data)
+        rslt = funcs.mad_std(data)
+        # want a scalar result, NOT a masked array
+        assert np.isscalar(rslt)
+
+        data[5,5] = np.nan
+        rslt = funcs.mad_std(data, ignore_nan=True)
+        assert np.isscalar(rslt)
+        with catch_warnings() as warns:
+            rslt = funcs.mad_std(data)
+            if NUMPY_GE_1_13:
+                # for recent numpy versions, numpy masked arrays appear to
```

Comment:
@keflavich -- I know the behavior is correct in 1.10.2, and incorrect in 1.11.2, and must still be in 12.0b1. 

It is concerning that it returns the incorrect value. Should an explicit check be made in `median_absolute_deviation` in the case of masked input and `ignore_nan=False` for something like `np.isnan(data).any()` and immediately return `nan` if that is the case?